core/src/test/java/org/apache/iceberg/TestManifestReader.java

```diff
@@ -145,4 +145,67 @@ public void testDeleteFilePositions() throws IOException {
       }
     }
   }
+
+  @Test
+  public void testReadDataManifestTable() throws IOException {
```

Comment:
I don't think that these tests should be in this class. This suite tests the manifest reader, not the manifest metadata tables. Why not put these tests in `TestIcebergSourceTablesBase`?

Reply:
I think this [test ](https://github.com/apache/iceberg/pull/2510/files#diff-9bb9188f62743b673504bdf41abf06d378ccb49443fd9828b0b65e88cdde7022R720) has covered the use cases.